molpipeline/experimental/custom_filter.py

```diff
@@ -0,0 +1,103 @@
+"""Module for custom filter functionality."""
+
+from __future__ import annotations
+
+from typing import Any, Callable, Optional
+
+try:
+    from typing import Self  # type: ignore[attr-defined]
+except ImportError:
+    from typing_extensions import Self
+
+from molpipeline.abstract_pipeline_elements.core import InvalidInstance
+from molpipeline.abstract_pipeline_elements.core import (
+    MolToMolPipelineElement as _MolToMolPipelineElement,
+)
+from molpipeline.utils.molpipeline_types import OptionalMol, RDKitMol
+
+
+class CustomFilter(_MolToMolPipelineElement):
+    """CustomFilter with bool functions on mol."""
+
+    def __init__(
+        self,
+        func: Callable[[RDKitMol], bool],
+        name: str = "CustomFilter",
+        n_jobs: int = 1,
+        uuid: Optional[str] = None,
+    ) -> None:
+        """Initialize CustomFilter.
+
+        Parameters
+        ----------
+        func : Callable[[RDKitMol], bool]
+            custom function to filter molecules
+        name : str, optional
+            name of the element, by default "CustomFilter"
+        n_jobs : int, optional
+            number of jobs to use, by default 1
+        uuid : str, optional
+            uuid of the element, by default None
+        """
+        super().__init__(name=name, n_jobs=n_jobs, uuid=uuid)
+        self.func = func
+
+    def pretransform_single(self, value: RDKitMol) -> OptionalMol:
+        """Pretransform single value.
```

Comment:
A bit more documentation would be nice

Reply:
For example:

"""Pretransform single value.

Applies the custom boolean function to the molecule.

...
"""